expressible/src/main/java/panda/std/stream/RandomComparator.java

```diff
@@ -0,0 +1,35 @@
+package panda.std.stream;
+
+import java.util.Comparator;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Original code from https://stackoverflow.com/a/35846296
+ */
+public final class RandomComparator<T> implements Comparator<T> {
```

Comment:
I'm not sure if that's the best option, because it's not working properly with duplicates in stream. E.g. this would work better:

* https://stackoverflow.com/a/40380283/3426515

Because it'd compare objects by `==`, not `equals`. Unfortunately, for smaller auto-boxed ints it still won't work. To avoid such issues, we should shuffle on collection. I think we should do sth like this:

* https://stackoverflow.com/a/35512753/3426515

and support both methods as:

```
stream.shuffle(ShuffleType.COPY) // default type, copy stream to list, shuffle and return new stream
stream.shuffle(ShuffleType.IN_PLACE) // with identity comparator
```